config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java

```diff
@@ -71,6 +74,7 @@ public DeploymentSpec(Optional<String> globalServiceId, UpgradePolicy upgradePol
         this.athenzDomain = athenzDomain;
         this.athenzService = athenzService;
         this.notifications = notifications;
+        this.endpoints = Objects.requireNonNull(endpoints, "Missing endpoints parameter");
```

Comment:
Constructor should throw if `endpoints` reference regions not contained in `steps`.